src/magic_spell_effect.cpp

```diff
@@ -764,6 +766,114 @@ static void spell_move( const spell &sp, const Creature &caster,
+    tripoint field_position = tripoint();
+
+    bool did_field_removal = false;
+
+    for( const auto &node : expander.area ) {
+        if( node.from == node.position ) {
+            continue;
+        }
+
+        field &target_field = here.field_at( node.position );
+        for( const std::pair<const field_type_id, field_entry> &fd : target_field ) {
+            if( fd.first.is_valid() && !fd.first.id().is_null() &&
+                fd.second.get_field_type() == target_field_type_id ) {
+                field_removed = target_field;
+                field_position = node.position;
+                here.remove_field( node.position, target_field_type_id );
+                did_field_removal = true;
+            }
+        }
+
+        if( did_field_removal ) {
+            // only remove one field in case of multiple
+            break;
+        }
+    }
+
+    return std::pair<field, tripoint> {field_removed, field_position};
+}
+
+static void handle_remove_fd_fatigue_field( const std::pair<field, tripoint> &fd_fatigue_field,
+        Creature &caster )
+{
+    for( const std::pair<const field_type_id, field_entry> &fd : std::get<0>( fd_fatigue_field ) ) {
+        const int &intensity = fd.second.get_field_intensity();
+        const translation &intensity_name = fd.second.get_intensity_level().name;
+        const tripoint &field_position = std::get<1>( fd_fatigue_field );
+        const bool sees_field = caster.sees( field_position );
+
+        switch( intensity ) {
+            case 1:
+
+                if( sees_field ) {
+                    caster.add_msg_if_player( m_neutral, _( "The %s fades.  You feel strange." ), intensity_name );
+                } else {
+                    caster.add_msg_if_player( m_neutral, _( "You suddenly feel strange." ) );
+                }
+
+                caster.add_effect( effect_teleglow, 1_hours );
+                break;
+            case 2:
+                if( sees_field ) {
+                    caster.add_msg_if_player( m_neutral, _( "The %s dissipates.  You feel strange." ), intensity_name );
+                } else {
+                    caster.add_msg_if_player( m_neutral, _( "You suddenly feel strange." ) );
+                }
+
+                caster.add_effect( effect_teleglow, 5_hours );
+                break;
+            case 3:
+                std::string message_prefix = "A nearby";
+
+                if( caster.sees( field_position ) ) {
+                    message_prefix = "The";
+                }
+
+                caster.add_msg_if_player( m_bad,
+                                          _( "%s %s pulls you in as it closes and ejects you violently!" ),
+                                          message_prefix, intensity_name );
+                caster.as_character()->hurtall( 10, nullptr );
+                caster.add_effect( effect_teleglow, 630_minutes );
+                teleport::teleport( caster );
+                break;
+        }
+        break;
```

Comment:
This `break` seems off here - it would break `for` loop unconditionally very after first iteration.

Reply:
Breaking unconditionally after the first iteration makes sense to me here.. at this point, we know for sure we have removed a single field and need to handle the removal. I can't recall why this uses a loop at all, there should only ever be one element. Been a while though.